### ISSUE TITLE
Tooling for bisecting crate versions to find violation ranges

### ISSUE THREAD

**Issue by djkoloski** [mozilla/cargo-vet#452]
When we find criteria violations in a crate, we need to record a version range of when the crate was affected. It can be difficult to track down when the violation was introduced because git history typically does not track which commits were included in a published crate version. The most reliable method for us to do this is to bisect published crate versions until we locate the version which first introduced the violation. Doing this manually can take a long time and is difficult for those not familiar with crates.io. Since cargo-vet already has support for inspecting crate sources at a particular version, it would be nice to include a feature to bisect the crate versions on top of the source inspection feature.

Split off from #446.

**Comment by bholley** [comment]
So it seems straightforward enough to build bisection tooling, but per the discussion in #446 I think the high-order bit is that the bisections in this workflow are unnecessarily eager, and we could largely eliminate the burden by deferring it until when it's actually relevant.

In other words, I think ideally we'd have a setup where you could record a violation in a specific version, and the tooling would conservatively assume it exists in all versions of the crate until a version is specified.

A simple way to do this today would just be to record the violation version broadly (`*`, or `>LAST_KNOWN_GOOD_VERSION`), record what we actually know in the notes, and then go back and update the audit entry if/when more specificity is desired. This depends on being able to actually update the audit record, so it's not very workable in a cross-organization context, but it might work for @djkoloski's use-case. WDYT?

**Comment by djkoloski** [comment]
Yes, I think this would work for our use case and obviate the need for tooling to bisect the exact offending version
